### PR TITLE
Avoid random() % 0 undefined behaviour when cluster-node-timeout < 30

### PR DESCRIPTION
**Pull request by enjoy-binbin**
Since #2449 made the failover delay relative to cluster-node-timeout.
Now delay = min(cluster-node-timeout / 30, 500), any cluster-node-timeout
below 30, including the legal minimum 0 will collapses delay to zero,
and `x % 0` is undefined behaviour.

### CODE REVIEW

**Review by sarthakaggarwal97 — APPROVED** [review]
Nice quick fix!

File: tests/unit/cluster/failover.tcl [hunk 1]
```diff
@@ -257,3 +257,30 @@ start_cluster 5 5 {tags {external:skip cluster}} {
...
+        resume_process [srv 0 pid]
+    }
+}
+
+start_cluster 3 1 {tags {external:skip cluster}} {
+    test_small_timeout 30
+} ;# start_cluster
+
+start_cluster 3 1 {tags {external:skip cluster}} {
+    test_small_timeout 20
```
Comment (enjoy-binbin) [inline tests/unit/cluster/failover.tcl]: Some OCD issues.
```suggestion
    test_small_timeout 15
```

File: tests/unit/cluster/failover.tcl [hunk 2]
```diff
@@ -257,3 +257,30 @@ start_cluster 5 5 {tags {external:skip cluster}} {
...
+    test_small_timeout 30
+} ;# start_cluster
+
+start_cluster 3 1 {tags {external:skip cluster}} {
+    test_small_timeout 15
+} ;# start_cluster
+
+start_cluster 3 1 {tags {external:skip cluster}} {
+    test_small_timeout 0
+} ;# start_cluster
```
Comment (zuiderkwast) [inline tests/unit/cluster/failover.tcl:282]: Why do we need to test 30 and 15? Test with 0 is enough?

`start_cluster` takes some time and we can speed up the tests if we skip some of them...

**Review by zuiderkwast — APPROVED** [review]

File: tests/unit/cluster/failover.tcl (hunk 2 above)
Comment (enjoy-binbin) [reply tests/unit/cluster/failover.tcl:282]: 0 is enough. I am just want to make sure we can cover more cases explicitly (some boundary values).
```
delay = min(cluster-node-timeout / 30, 500)
```

Comment (enjoy-binbin) [reply tests/unit/cluster/failover.tcl:282]: The three tests will take 24s in local.

Comment (zuiderkwast) [reply tests/unit/cluster/failover.tcl:282]: 24s is a lot of time. We are adding more test cases all the time. Let's add only 2 instead of 3?

File: tests/unit/cluster/failover.tcl [hunk 3]
```diff
@@ -257,3 +257,30 @@ start_cluster 5 5 {tags {external:skip cluster}} {
...
+    }
+}
+
+start_cluster 3 1 {tags {external:skip cluster}} {
+    test_small_timeout 30
+} ;# start_cluster
+
+start_cluster 3 1 {tags {external:skip cluster}} {
+    test_small_timeout 15
+} ;# start_cluster
```
Comment (enjoy-binbin) [inline tests/unit/cluster/failover.tcl]:
```suggestion
```

File: tests/unit/cluster/failover.tcl (hunk 2 above)
Comment (enjoy-binbin) [reply tests/unit/cluster/failover.tcl:282]: sure. i am removing the 15 one.
